### ISSUE TITLE
False positive: Submit button in a component with a `<form>` not detected

### ISSUE THREAD

**Issue by fdeters** [johanrd/html-validate-ember#34]
## Description

I'm seeing a false positive when a component has a `<form>` element with a `<button type="submit">`. The invocation of the component throws a [`wcag/h32`](https://html-validate.org/rules/wcag/h32.html) violation despite the valid structure.

For example:

```gjs
// app/components/form-with-submit.gjs

<template>
  <form>
    <label for="foo">Foo</label>
    <input type="text" name="foo" id="foo" />

    <button type="submit">Submit</button>
  </form>
</template>
```
```gjs
// app/templates/application.gjs

import FormWithSubmit from '../components/form-with-submit';

<template>
  {{! this form has a submit button but the invocation here throws wcag/h32 }}
  <FormWithSubmit />

  {{outlet}}
</template>
```

## Reproduction Sample

https://github.com/fdeters/html-validate-ember-list-false-positive
